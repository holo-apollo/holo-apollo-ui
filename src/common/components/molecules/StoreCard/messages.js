import { defineMessages } from 'react-intl';

export default defineMessages({
  goods: {
    id: 'StoreCard.goods',
    defaultMessage: `{goodsCount, plural, 
      one {good}
      few {goods}
      many {goods}
      other {goods}
    }`,
  },
});
