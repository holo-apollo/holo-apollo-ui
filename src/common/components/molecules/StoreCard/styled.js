// @flow
import styled, { css } from 'styled-components';

import palette from 'common/palette';

export const Cont = styled.div`
  color: ${palette.grey};
`;

const getCoverContStyles = ({
  isSmall,
  imgUrl,
  hovered,
}: {
  isSmall: boolean,
  imgUrl: string,
  hovered: boolean,
}) => css`
  height: ${isSmall ? 100 : 400}px;
  background-image: url(${imgUrl});
  border: solid 1px ${isSmall && hovered ? palette.darkGrey : palette.lightGrey};
`;

export const CoverCont = styled.div`
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
  ${getCoverContStyles};
`;

export const BottomCont = styled.div`
  display: flex;
`;

const getContentContStyles = ({
  isSmall,
  hovered,
}: {
  isSmall: boolean,
  hovered: boolean,
}) => css`
  border: solid 1px ${isSmall && hovered ? palette.darkGrey : palette.lightGrey};
  border-top: none;
`;

export const ContentCont = styled.div`
  display: flex;
  position: relative;
  min-height: 200px;
  flex: 1;
  ${getContentContStyles};
`;

export const BannerCont = styled.div`
  flex: 1;
`;

export const getAvatarContStyles = ({
  isSmall,
  imgUrl,
}: {
  isSmall: boolean,
  imgUrl: string,
}) => css`
  width: ${isSmall ? 100 : 150}px;
  height: ${isSmall ? 100 : 150}px;
  left: ${isSmall ? 35 : 23}px;
  background-image: url(${imgUrl});
`;

export const AvatarCont = styled.div`
  border-radius: 50%;
  border: solid 3px ${palette.darkGrey};
  box-sizing: border-box;
  position: absolute;
  top: 0;
  transform: translateY(-50%);
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
  ${getAvatarContStyles};
`;

export const getContentLeftContStyles = ({
  isSmall,
  hovered,
}: {
  isSmall: boolean,
  hovered: boolean,
}) => css`
  padding: ${isSmall ? 35 : 23}px;
  border-right: solid 1px
    ${isSmall && hovered ? palette.darkGrey : palette.lightGrey};
`;

export const ContentLeftCont = styled.div`
  flex-basis: 70%;
  ${getContentLeftContStyles};
`;

export const ContentRightCont = styled.div`
  flex-basis: 30%;
  display: flex;
  flex-direction: column;
`;

const getGoodsCountContStyles = ({
  isSmall,
  hovered,
}: {
  isSmall: boolean,
  hovered: boolean,
}) => css`
  border-bottom: 1px solid
    ${isSmall && hovered ? palette.darkGrey : palette.lightGrey};
`;

export const GoodsCountCont = styled.div`
  flex: 1;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 12px;
  ${getGoodsCountContStyles};
`;

export const RatingCont = styled.div`
  flex: 1;
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const GoodsCountWrapper = styled.span`
  display: flex;
  align-items: baseline;
`;

export const GoodsCount = styled.span`
  font-size: 24px;
  margin-right: 5px;
`;

export const RatingWrapper = styled.span`
  display: flex;
  align-items: center;
`;

export const Rating = styled.span`
  font-size: 24px;
  margin-left: 5px;
`;

const getMainInfoContStyles = ({ isSmall }: { isSmall: boolean }) => css`
  margin-top: ${isSmall ? 30 : 0}px;
  margin-left: ${isSmall ? 0 : 164}px;
  margin-bottom: ${isSmall ? 12 : 33}px;
`;

export const MainInfoCont = styled.div`
  ${getMainInfoContStyles};
`;

export const StoreNameCont = styled.div`
  font-size: 18px;
  font-weight: 500;
  margin-bottom: 8px;
`;

export const LocationCont = styled.div`
  font-size: 12px;
`;

export const DescriptionCont = styled.div`
  font-size: 12px;
`;
