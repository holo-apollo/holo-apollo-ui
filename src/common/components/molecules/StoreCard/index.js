// @flow
import React from 'react';
import { FormattedMessage } from 'react-intl';
import { Emoji } from 'emoji-mart';

import {
  Cont,
  CoverCont,
  BottomCont,
  ContentCont,
  BannerCont,
  AvatarCont,
  ContentLeftCont,
  ContentRightCont,
  GoodsCountCont,
  RatingCont,
  GoodsCountWrapper,
  GoodsCount,
  RatingWrapper,
  Rating,
  MainInfoCont,
  StoreNameCont,
  LocationCont,
  DescriptionCont,
} from './styled';
import messages from './messages';

type Props = {
  isSmall: boolean,
  storeName: string,
  location: string,
  description: string,
  goodsCount: number,
  coverImageUrl: string,
  avatarUrl: string,
  rating?: number,
};

const StoreCard = ({
  isSmall,
  coverImageUrl,
  avatarUrl,
  goodsCount,
  rating,
  storeName,
  location,
  description,
}: Props) => (
  <Cont>
    <CoverCont isSmall={isSmall} imgUrl={coverImageUrl} />
    <BottomCont>
      <ContentCont>
        <AvatarCont isSmall={isSmall} imgUrl={avatarUrl} />
        <ContentLeftCont isSmall={isSmall}>
          <MainInfoCont isSmall={isSmall}>
            <StoreNameCont>{storeName}</StoreNameCont>
            <LocationCont>{location}</LocationCont>
          </MainInfoCont>
          <DescriptionCont>{description}</DescriptionCont>
        </ContentLeftCont>
        <ContentRightCont>
          <GoodsCountCont>
            <GoodsCountWrapper>
              <GoodsCount>{goodsCount}</GoodsCount>
              <FormattedMessage {...messages.goods} values={{ goodsCount }} />
            </GoodsCountWrapper>
          </GoodsCountCont>
          <RatingCont>
            <RatingWrapper>
              <Emoji emoji="blue_heart" size={24} />
              <Rating>{rating || '-'}</Rating>
            </RatingWrapper>
          </RatingCont>
        </ContentRightCont>
      </ContentCont>
      {!isSmall && <BannerCont />}
    </BottomCont>
  </Cont>
);

export default StoreCard;
