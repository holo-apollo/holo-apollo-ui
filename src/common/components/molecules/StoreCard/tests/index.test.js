import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import StoreCard from '../index';

describe('StoreCard', function() {
  it('should render normally', function() {
    const wrapper = mount(
      <StoreCard
        isSmall={false}
        storeName="Armonie"
        location="Lviv"
        description={
          'Scented candles from natural soy wax. ' +
          'Perfumed with the best essential oils. ' +
          'Fill your home with a cozy aroma from Armonie.'
        }
        goodsCount={435}
        coverImageUrl="http://www.trendycovers.com/covers/abstract_3d_facebook_cover_1370594397.jpg"
        avatarUrl="http://flowersandmorebyerin.com/wp-content/uploads/2015/02/pFlower3.png"
        rating={4.8}
      />
    );
    expect(wrapper.exists()).toBe(true);
  });

  it('should match snapshot if small', function() {
    const tree = renderer
      .create(
        <StoreCard
          isSmall={true}
          storeName="Armonie"
          location="Lviv"
          description={
            'Scented candles from natural soy wax. ' +
            'Perfumed with the best essential oils. ' +
            'Fill your home with a cozy aroma from Armonie.'
          }
          goodsCount={435}
          coverImageUrl="http://www.trendycovers.com/covers/abstract_3d_facebook_cover_1370594397.jpg"
          avatarUrl="http://flowersandmorebyerin.com/wp-content/uploads/2015/02/pFlower3.png"
          rating={4.8}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot if big', function() {
    const tree = renderer
      .create(
        <StoreCard
          isSmall={false}
          storeName="Armonie"
          location="Lviv"
          description={
            'Scented candles from natural soy wax. ' +
            'Perfumed with the best essential oils. ' +
            'Fill your home with a cozy aroma from Armonie.'
          }
          goodsCount={435}
          coverImageUrl="http://www.trendycovers.com/covers/abstract_3d_facebook_cover_1370594397.jpg"
          avatarUrl="http://flowersandmorebyerin.com/wp-content/uploads/2015/02/pFlower3.png"
          rating={4.8}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
