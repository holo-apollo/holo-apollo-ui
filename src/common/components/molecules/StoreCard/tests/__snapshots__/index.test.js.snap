// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreCard should match snapshot if big 1`] = `
<div
  className="styled__Cont-u4l7vz-0 crbnux"
>
  <div
    className="styled__CoverCont-u4l7vz-1 ewSHZq"
  />
  <div
    className="styled__BottomCont-u4l7vz-2 icAjlA"
  >
    <div
      className="styled__ContentCont-u4l7vz-3 AHQbF"
    >
      <div
        className="styled__AvatarCont-u4l7vz-5 bQxxaf"
      />
      <div
        className="styled__ContentLeftCont-u4l7vz-6 dzlxSF"
      >
        <div
          className="styled__MainInfoCont-u4l7vz-14 dOJmPc"
        >
          <div
            className="styled__StoreNameCont-u4l7vz-15 ekQkWm"
          >
            Armonie
          </div>
          <div
            className="styled__LocationCont-u4l7vz-16 dSwJMW"
          >
            Lviv
          </div>
        </div>
        <div
          className="styled__DescriptionCont-u4l7vz-17 hmBFUp"
        >
          Scented candles from natural soy wax. Perfumed with the best essential oils. Fill your home with a cozy aroma from Armonie.
        </div>
      </div>
      <div
        className="styled__ContentRightCont-u4l7vz-7 xNeLC"
      >
        <div
          className="styled__GoodsCountCont-u4l7vz-8 cSrIku"
        >
          <span
            className="styled__GoodsCountWrapper-u4l7vz-10 ffxBJU"
          >
            <span
              className="styled__GoodsCount-u4l7vz-11 laLMj"
            >
              435
            </span>
            <span>
              goods
            </span>
          </span>
        </div>
        <div
          className="styled__RatingCont-u4l7vz-9 XiMlu"
        >
          <span
            className="styled__RatingWrapper-u4l7vz-12 fshdtH"
          >
            <span
              className="emoji-mart-emoji"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              title={null}
            >
              <span
                style={
                  Object {
                    "backgroundImage": "url(https://unpkg.com/emoji-datasource-apple@4.0.4/img/apple/sheets-256/64.png)",
                    "backgroundPosition": "47.05882352941176% 100%",
                    "backgroundSize": "5200% 5200%",
                    "display": "inline-block",
                    "height": 24,
                    "width": 24,
                  }
                }
              />
            </span>
            <span
              className="styled__Rating-u4l7vz-13 gCLjQd"
            >
              4.8
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      className="styled__BannerCont-u4l7vz-4 bDVjaq"
    />
  </div>
</div>
`;

exports[`StoreCard should match snapshot if small 1`] = `
<div
  className="styled__Cont-u4l7vz-0 crbnux"
>
  <div
    className="styled__CoverCont-u4l7vz-1 elzTxL"
  />
  <div
    className="styled__BottomCont-u4l7vz-2 icAjlA"
  >
    <div
      className="styled__ContentCont-u4l7vz-3 AHQbF"
    >
      <div
        className="styled__AvatarCont-u4l7vz-5 dwYyBI"
      />
      <div
        className="styled__ContentLeftCont-u4l7vz-6 bUlPXO"
      >
        <div
          className="styled__MainInfoCont-u4l7vz-14 lhgUJl"
        >
          <div
            className="styled__StoreNameCont-u4l7vz-15 ekQkWm"
          >
            Armonie
          </div>
          <div
            className="styled__LocationCont-u4l7vz-16 dSwJMW"
          >
            Lviv
          </div>
        </div>
        <div
          className="styled__DescriptionCont-u4l7vz-17 hmBFUp"
        >
          Scented candles from natural soy wax. Perfumed with the best essential oils. Fill your home with a cozy aroma from Armonie.
        </div>
      </div>
      <div
        className="styled__ContentRightCont-u4l7vz-7 xNeLC"
      >
        <div
          className="styled__GoodsCountCont-u4l7vz-8 cSrIku"
        >
          <span
            className="styled__GoodsCountWrapper-u4l7vz-10 ffxBJU"
          >
            <span
              className="styled__GoodsCount-u4l7vz-11 laLMj"
            >
              435
            </span>
            <span>
              goods
            </span>
          </span>
        </div>
        <div
          className="styled__RatingCont-u4l7vz-9 XiMlu"
        >
          <span
            className="styled__RatingWrapper-u4l7vz-12 fshdtH"
          >
            <span
              className="emoji-mart-emoji"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              title={null}
            >
              <span
                style={
                  Object {
                    "backgroundImage": "url(https://unpkg.com/emoji-datasource-apple@4.0.4/img/apple/sheets-256/64.png)",
                    "backgroundPosition": "47.05882352941176% 100%",
                    "backgroundSize": "5200% 5200%",
                    "display": "inline-block",
                    "height": 24,
                    "width": 24,
                  }
                }
              />
            </span>
            <span
              className="styled__Rating-u4l7vz-13 gCLjQd"
            >
              4.8
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
